src/components/header.jsx

```diff
@@ -55,12 +55,11 @@ function Header({ slug }) {
   );
 
   const handleClick = (langKey) => {
-    i18n.changeLanguage(langKey);
-    navigate(`/${langKey}/${slug}`);
+    i18n.changeLanguage(langKey, navigate(`/${langKey}/${slug}`));
```

Comment:
Have you tried changing the language to see if this works?

Reply:
yes